src/app/Hooks/components/helpers.ts

```diff
@@ -0,0 +1,45 @@
+import { CLUSTER_API_VERSION, META } from '@app/common/constants';
+import { IHook } from '@app/queries/types';
+import React from 'react';
+import { HookFormState } from './AddEditHookModal';
+
+export const generateHook = (form: HookFormState): IHook => ({
+  apiVersion: CLUSTER_API_VERSION,
+  kind: 'Hook',
+  metadata: {
+    name: form.values.name,
+    namespace: META.namespace,
+  },
+  spec: {
+    url: form.values.url,
+    branch: form.values.branch,
+  },
+});
+
+interface IEditHookPrefillEffect {
+  isDonePrefilling: boolean;
+}
+
+export const useEditHookPrefillEffect = (
```

Comment:
Nice work getting all this scaffolding in. We definitely will need all this prefilling stuff, etc. Although since this prefill doesn't require waiting for any queries, we may not need the `isDonePrefilling` logic... eh. We can simplify later, there's probably some other reason I did that. 🙃

Reply:
Thanks.